pkg/downloader/chart_downloader.go

```diff
@@ -94,12 +95,31 @@ func (c *ChartDownloader) DownloadTo(ref, version, dest string) (string, *proven
 		return "", nil, err
 	}
 
-	data, err := g.Get(u.String(), c.Options...)
+	downloadURL := u.String()
+	name := filepath.Base(u.Path)
+
+	if _, ok := g.(*registry.Getter); ok {
+		parts := strings.Split(filepath.Base(u.Path), ":")
+
+		if len(parts) == 1 && version == "" {
+			return "", nil, errors.New("no version or tag provided")
+		}
+
+		if len(parts) != 2 {
```

Comment:
If I'm reading this right, the following will append a version to the path if there are multiple colons in the URL path, like so:

```
repository: oci://localhost:5000/testrepo/testchart:1.2.3:1.2.3
version: 1.0.0
```

Which will result in the following URL path: `/testrepo/testchart:1.2.3:1.2.3:1.0.0`

Is that right? That doesn't seem like valid input.

Reply:
Multiple colons in the path isn't allowed; you can't tag an OCI image with `localhost:5000/testrepo/testchart:1.2.3:1.2.3`;

```sh
➜ docker tag nginx:1.16.0 localhost:5000/nginx:1.16.0:1.2.3
Error parsing reference: "localhost:5000/nginx:1.16.0:1.2.3" is not a valid repository/tag: invalid reference format
```

Reply:
Sorry for barging in, but isn't this shorter & more correct:
```
		if len(parts) == 1 {
			if version == "" {
				return "", nil, errors.New("no version or tag provided")
			}

			parts = append(parts, version)
			u.Path = fmt.Sprintf("%s:%s", u.Path, version)
		}
```

You wouldn't be doing `len()` twice and would only be appending version if `:` is not present in the path at all.